apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -80,12 +113,19 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
 
         // Reintroduce color (Connelly: use eps to avoid scaling up noise w/ apollo3.png input)
         Func color;
-        float eps = 0.01f;
-        color(x, y, c) = outGPyramid[0](x, y) * (floating(x, y, c)+eps) / (gray(x, y)+eps);
+        if (float_point) {
+            float eps = 0.01f;
+            color(x, y, c) = outGPyramid[0](x, y) * (floating(x, y, c)+eps) / (gray(x, y)+eps);
 
-        // Convert back to 16-bit
-        output(x, y, c) = cast<uint16_t>(clamp(color(x, y, c), 0.0f, 1.0f) * 65535.0f);
+            // Convert back to 16-bit
+            output(x, y, c) = cast<uint16_t>(clamp(color(x, y, c), 0.0f, 1.0f) * 65535.0f);
+        } else {
 
+            color(x, y, c) = floating(x, y, c) + (outGPyramid[0](x, y) - gray(x, y)) / 4;
```

Comment:
This looks like a really big change: the old code computed GPyramid * output / input, this is computing output + (GPyramid - input). At the very least, this needs a comment pointing this out/explaining.